pkg/vtpm/src/server.cpp

```diff
@@ -464,28 +547,11 @@ main (int argc, char *argv[])
             cerr << "Error in accepting client connection" << std::endl;
         }
 
-        ssize_t recvBytes = 0;
-        char hdrBuf[CODED_STRM_HDR_LEN];
-        char *pBuf = hdrBuf;
-
         cout << "New connection from: " << inet_ntoa(clientAddr.sin_addr)
-            << std::endl;
+             << ":" << to_string(clientAddr.sin_port) << std::endl;
 
-        recvBytes = recv(sock, pBuf, CODED_STRM_HDR_LEN, MSG_PEEK);
-        if (recvBytes < 0) {
-          int rc = errno;
-          cerr << "recv() from " << inet_ntoa(clientAddr.sin_addr)
-               << " failed with " << rc << std::endl;
-        } else if (recvBytes == CODED_STRM_HDR_LEN) {
-            if (handleRequest(sock, readHdr(hdrBuf)) != 0) {
-                cerr << "Failure processing the request" << std::endl;
-            }
-        } else {
-           cerr << "recv() received fewer than expected(" << recvBytes
-                << ") bytes from: " << inet_ntoa(clientAddr.sin_addr)
-                << std::endl;
-        }
-        close(sock);
+        std::thread worker(serveClient, sock, clientAddr);
+        worker.detach();
```

Comment:
Do you need to do something to pick up the exit status from the threads?

Reply:
Right now there is no need to pick up the exit status, since the threads need to run to completion, 
and exit in one form or the other. Success or a failure(in which case the thread itself takes care of 
sending error messages to the client and/or logging it)

Reply:
OK.
Makes sense to stress test to make sure there is no memory leakage due to threads in the server with 1 million requests.

Reply:
I did the following test:

Had a simple command that takes one input file name and one output filename. 
Built a test container with server and client on the same container. 
Wrote a shell script to invoke "eve_run test_cmd -i <input> -o <output>" 1 million times
Took /proc/x/status of the server before and after the exercise, and also the thread counts
Didn't see any memory leakage in the outputs.
The Dockerfile, test scripts, test commands, and the scale logs can be found here:
https://github.com/cshari-zededa/vtpm_scale_test/tree/master/scale_test

Reply:
Great